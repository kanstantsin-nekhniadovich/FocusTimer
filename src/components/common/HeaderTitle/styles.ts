import { StyleSheet } from 'react-native';
import { Typography } from '@styles';

export const styles = StyleSheet.create({
  title: {
    ...Typography.h3,
    fontSize: 18,
  },
});
