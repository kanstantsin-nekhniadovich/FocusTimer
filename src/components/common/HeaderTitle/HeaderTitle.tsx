import React from 'react';
import { Text, StyleProp, TextStyle } from 'react-native';

import { styles } from './styles';

interface Props {
  title: string;
  style?: StyleProp<TextStyle>;
}

export const HeaderTitle: React.FC<Props> = ({ title, style }) =>
  <Text numberOfLines={1} style={[styles.title, style]}>{title}</Text>;
