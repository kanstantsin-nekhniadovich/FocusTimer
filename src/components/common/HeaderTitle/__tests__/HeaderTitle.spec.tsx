import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';

import { HeaderTitle } from '../HeaderTitle';

const title = 'Test title';

const renderComponent = (): RenderAPI => {
  return render(<HeaderTitle title={title} />);
};

describe('HeaderTitle', () => {
  it('should match snapshot', () => {
    const component = renderComponent();
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render HeaderTitle with passed title', () => {
    const { getByText } = renderComponent();

    expect(getByText(title)).toBeDefined();
  });
});
