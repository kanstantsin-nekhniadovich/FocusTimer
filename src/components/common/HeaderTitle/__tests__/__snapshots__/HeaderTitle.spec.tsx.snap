// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle should match snapshot 1`] = `
<Text
  numberOfLines={1}
  style={
    Object {
      "fontFamily": "DMSans_500Medium",
      "fontSize": 20,
      "letterSpacing": 0,
      "lineHeight": 26,
    }
  }
>
  Test title
</Text>
`;
